### PR TITLE
[ActivityIndicator] - Added voice accessibility support for MDCActivityIndicator

### PR DESCRIPTION
**Pull request by ShepJGoogle**
Added accessibility to MDCActivityIndicator

Accessibility Label:
- Returns "In progress" for indeterminate mode isSpinning
- Returns "{value} Percent Complete" for determinate mode isSpinning
- Returns "Progress Halted" for both modes determinate and indeterminate

Accessibility Value:
- Returns "1" for in progress
- Returns "0" for not in progress

Values From internal issue: b/27796425

### CODE REVIEW

**Comment by ShepJGoogle** [comment]
Open to suggestions to change values or labels while waiting for accessibility advice

File: components/ActivityIndicator/src/MDCActivityIndicator.m [hunk 1]
```diff
@@ -809,4 +814,73 @@ - (void)applyPropertiesWithoutAnimation:(void (^)(void))setPropBlock {
...
+  return YES;
+}
+
+- (NSString *)accessibilityLabel {
+
+  if (self.isAnimating) {
+    if (self.indicatorMode == MDCActivityIndicatorModeIndeterminate) {
+      NSString *key =
+      kMaterialActivityIndicatorStringTable[kStr_MaterialActivityIndicatorInProgressAccessibilityLabel];
+      return NSLocalizedStringFromTableInBundle(key,
```
Comment (romoore) [inline components/ActivityIndicator/src/MDCActivityIndicator.m:850]: If we don't localize the strings outside of English, and we're retrieving them from our bundle, can clients still provide translations?

File: components/ActivityIndicator/src/MDCActivityIndicator.m [hunk 2]
```diff
@@ -809,4 +814,73 @@ - (void)applyPropertiesWithoutAnimation:(void (^)(void))setPropBlock {
...
+  }
+}
+
+- (UIAccessibilityTraits)accessibilityTraits {
+  return UIAccessibilityTraitUpdatesFrequently;
+}
+
+- (NSString *)accessibilityValue {
+  if (self.isAnimating) {
+    return @"1";
```
Comment (romoore) [inline components/ActivityIndicator/src/MDCActivityIndicator.m]: Should this be the localized value of the number 1 or should it always be an Arabic numeral "1" as a string?

File: components/ActivityIndicator/src/MDCActivityIndicator.m [hunk 3]
```diff
@@ -809,4 +814,73 @@ - (void)applyPropertiesWithoutAnimation:(void (^)(void))setPropBlock {
...
+  } else {
+    return @"0";
```
Comment (romoore) [inline components/ActivityIndicator/src/MDCActivityIndicator.m]: Same for "0"

File: components/ActivityIndicator/src/MDCActivityIndicator.m (hunk 1 above)
Comment (ShepJGoogle) [reply components/ActivityIndicator/src/MDCActivityIndicator.m:850]: Per @ianegordon request we wanted to get an initial english version out

Comment (romoore) [reply components/ActivityIndicator/src/MDCActivityIndicator.m:850]: What happens if the client wants to set an accessibility label? On a UIProgressView the default label is "Progress" and setting the accessibilityLabel works.

File: components/ActivityIndicator/src/MDCActivityIndicator.m [hunk 4]
```diff
@@ -809,4 +814,73 @@ - (void)applyPropertiesWithoutAnimation:(void (^)(void))setPropBlock {
...
+      NSString *key =
+      kMaterialActivityIndicatorStringTable[kStr_MaterialActivityIndicatorInProgressAccessibilityLabel];
+      return NSLocalizedStringFromTableInBundle(key,
+                                                kMaterialActivityIndicatorStringsTableName,
+                                                [[self class] bundle],
+                                                @"In Progress");
+    } else {
+      NSUInteger percentage = (int)(self.progress * 100);
+      NSString *key =
+      kMaterialActivityIndicatorStringTable[kStr_MaterialActivityIndicatorProgressCompletedAccessibilityLabel];
```
Comment (willlarche) [inline components/ActivityIndicator/src/MDCActivityIndicator.m:857]: +4

File: components/ActivityIndicator/src/MDCActivityIndicator.m [hunk 5]
```diff
@@ -809,4 +814,73 @@ - (void)applyPropertiesWithoutAnimation:(void (^)(void))setPropBlock {
...
+      NSString *localizedString = NSLocalizedStringFromTableInBundle(key,
+                                                kMaterialActivityIndicatorStringsTableName,
```
Comment (willlarche) [inline components/ActivityIndicator/src/MDCActivityIndicator.m:859]: Double check spacing with format_all.

File: components/ActivityIndicator/src/MaterialActivityIndicator.bundle/Resources/en.lproj/MaterialActivityIndicator.strings [hunk 6]
```diff
@@ -0,0 +1,8 @@
+/* Accessibility label for indeterminate activity indicator in progress */
+"MaterialActivityIndicatorInProgressAccessibilityLabel" = "In Progress";
```
Comment (willlarche) [inline components/ActivityIndicator/src/MaterialActivityIndicator.bundle/Resources/en.lproj/MaterialActivityIndicator.strings:2]: I think the words that aren't the start of each quote should be lowercase. Sentence case.

**Review by willlarche — APPROVED** [review]

File: components/ActivityIndicator/src/MDCActivityIndicator.m (hunk 1 above)
Comment (ajsecord) [reply components/ActivityIndicator/src/MDCActivityIndicator.m:850]: Rob, we're not expecting clients to provide translations, that's our responsibility.

Also, how would you suggest we return a _dynamic_ label without this code?

Comment (romoore) [reply components/ActivityIndicator/src/MDCActivityIndicator.m:850]: I think the dynamic label is great and we may want to mark the setter as not available, or something else (NSAssert) to let downstream dependencies know that setting the accessibilityLabel will have no effect.  Maybe we want to provide these as default values that clients can override.

**Comment by ianegordon** [comment]
@romoore has raised some concerns that we should investigate which I've captured in #1841.

I'm will merge as this implementation is an improvement over our existing behavior and any improvements we make will depend on the localization capabilities added here.

**Pull request merged**